tstest/integration/vms/ssh_test.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build linux
+
+package vms
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/terminal"
+	"tailscale.com/types/logger"
+)
+
+func mkSSHServer(t *testing.T, hostKey ssh.Signer, bindhost string) string {
```

Comment:
This is a surprising choice to me. ssh is a pain to use without a lot of setup. Do some of these systems ship without netcat or curl?

Reply:
Netcat is flag-inconsistent across distros and not in all of the base images. We do add curl for other reasons in the test stack, however SSH more closely represents how people use Tailscale.

Reply:
I don't understand what running an ssh server in this unusual mode, moving text in a way that, according to your comment isn't the way ssh is meant to be used, tests tailscale in any useful way that an HTTP POST from curl wouldn't. They are both, from Tailscale's perspective, typical TCP sockets.

If you want to submit this to get on with working on some other part of this system that's fine, but add a `TODO: replace with something simpler like curl` to the top.

Reply:
I made it use HTTP instead.